src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,25 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    public String value;
+    protected boolean isPrivate;
+    
+    public Contact(){
+        
```

Comment:
Perhaps you can extract another method from all the child classes that deals with the validation. The method will take in the regex string and perform the check.